README.md

```diff
@@ -192,67 +208,74 @@ If you don't want to use predefined ACLs you can define your own (the ACL object
-- Any established watches may/will be fired once each when/if your client is expired by the ZK server, the input arguments are observed to be: type=-1, state=1, path="". Care should be taken to handle this differently than a "real" watch event if that matters to your application.
+## Known Bugs & Issues
+
+* The lib will segfault if you try to use a ZooKeeper intance after the on_closed event is delivered (possibly as a result of session timeout etc.) YOU MAY NOT re-use the closed ZooKeeper instance. You should allocate a new one and initialize it as a completely new client. Any and all watchers from your first instance are lost, though they may fire (before the on_close) see below.
+* Any established watches may/will be fired once each when/if your client is expired by the ZK server, the input arguments are observed to be: type=-1, state=1, path="". Care should be taken to handle this differently than a "real" watch event if that matters to your application.
+
+## Contribute to the project
 
-# Contribute to the project #
 Check out the issues tab and grab one! Read the [contributing](./CONTRIBUTING.md) document.
 
-# See Also #
+## See Also
 
-- [http://zookeeper.apache.org/releases.html](http://zookeeper.apache.org/releases.html)
-- [http://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#ZooKeeper+C+client+API](http://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#ZooKeeper+C+client+API)
+* [http://zookeeper.apache.org/releases.html](http://zookeeper.apache.org/releases.html)
+* [http://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#ZooKeeper+C+client+API](http://zookeeper.apache.org/doc/current/zookeeperProgrammers.html#ZooKeeper+C+client+API)
 
-# Acknowledgments #
+## Acknowledgments
 
-- **[node-webworker](http://github.com/pgriess/node-webworker "node-webworker") by pgriess** is used to spawn multiple ZK workers in one of the tests.
+* **[node-webworker](http://github.com/pgriess/node-webworker "node-webworker") by pgriess** is used to spawn multiple ZK workers in one of the tests.
 
-# LICENSE #
+## LICENSE
 
 See [LICENSE-MIT.txt](./LICENSE-MIT.txt) file in the top level folder.
 
-# ORIGINAL AUTHOR #
+## ORIGINAL AUTHOR
 
 Yuri Finkelstein (yurif2003 at yahoo dot com)
 
 with awesome contributions from:
 
-- Woody Anderson (Woodya)
-- Dave Dopson (ddopson)
-- Ryan Phillips (rphillips)
-- David Trejo (DTrejo)
-- Mark Cavage (mcavage)
-- Jakub Lekstan (kuebk)
-- Matt Lavin (mdlavin)
-- David Vujic (davidvujic)
+* Woody Anderson (Woodya)
+* Dave Dopson (ddopson)
+* Ryan Phillips (rphillips)
+* David Trejo (DTrejo)
+* Mark Cavage (mcavage)
+* Jakub Lekstan (kuebk)
+* Matt Lavin (mdlavin)
+* David Vujic (davidvujic)
```

Comment:
In a separate PR, I will add your name to this list!

Reply:
Thank you! It's my first open source project that I contribute to on a regular basis. I must admit that you do a great job in pushing the development of this library!